tests/rules/parsing_method.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * @file Test for no-unsanitized rule
+ * @author Frederik Braun et al.
+ * @copyright 2015-2017 Mozilla Corporation. All rights reserved
+ */
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../lib/rules/parsing_method");
+const RuleTester = require("eslint").RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const eslintTester = new RuleTester();
+
+eslintTester.run("parsing_method", rule, {
+    // Examples of code that should not trigger the rule
+    // XXX this does not find z['innerHTML'] and the like.
+
+    valid: [
+        {
+            code: "var doc = DOMParser.parseFromString('static string');",
+        },
+        {
+            code: "var doc = Document.parseHTMLUnsafe('static string');",
+        },
+    ],
+
+    // Examples of code that should trigger the rule
+    invalid: [
+        /* XXX Do NOT change the error strings below without review from freddy:
+         * The strings are optimized for SEO and understandability.
+         * The developer can search for them and will find this MDN article:
+         *  https://developer.mozilla.org/en-US/Firefox_OS/Security/Security_Automation
+         */
+
+        {
+            code: "var doc = DOMParser.parseFromString(badness);",
+            errors: [
+                {
+                    message:
+                        /Unsafe call to DOMParser.parseFromString for argument 0/,
```

Comment:
That feels like a good point to be too, let me take a look to what additional change may be needed in this PR if we would like to use slightly different message for the separate parsing_method rule.

Reply:
@mozfreddyb I have included in [afe2df6](https://github.com/mozilla/eslint-plugin-no-unsanitized/pull/244/commits/afe2df60e3e1c7920b7b6e40967104601d227a88) one possible approach to customize the message (for now I went for the simpler "Potentially unsafe call..." one, but if the approach looks reasonable we can tweak the message further).

I have also changed the objectMatches for the DOMParser parseFromString method to 'parser' for now.

I was actually tempted to go even for just 'parse' or 'pars' (so that it would also match other variations, like 'htmlParsing' or 'parseHTML' or 'parseUtils'), wdyt?